### DISCUSSION TITLE
solar radiation

### DISCUSSION THREAD

**Discussion by sean9528** [open-meteo/open-meteo#119] [Q&A]
I know that the solar radiation of DWD is processed by moving average. I downloaded the data and verified it. After processing the data with moving averages, the first point is 0. I'm curious how did you reconstruct this data

**Comment by patrick-zippenfenig** [comment]
The first timestep for backwards averaged or accumulated variables like precipitation is always unavailable. In this case, it is just skipped and data form the previous run is used.

Did you get the same results as in the Open-Meteo API?

**Reply by sean9528** [reply]
I didn't get the same results as in the Open-Meteo API. I only download data occasionally，so I can't always get the data from the previous run.

**Reply by sean9528** [reply]
In fact, only the solar radiation seem to have been processed by moving averages

**Reply by sean9528** [reply]
> The first timestep for backwards averaged or accumulated variables like precipitation is always unavailable. In this case, it is just skipped and data form the previous run is used.
> 
> Did you get the same results as in the Open-Meteo API?

you really did a good job. This API is useful for those of us who have weather data needs. For me, processing the raw data is a hassle.

The first,  I'm retracting the answer that I didn't get the same results as in  Open-Meteo API. This answer is based on my mistaken understanding of moving averages.

Now I have a new question to ask you. I got some of the same results as in the Open-Meteo API, but some of them are different. In the DWD, the frequency of the last part is 3 hours. I tried to use spline interpolation method to restore the data, but the result is different as in the Open-Meteo API. The method of restoring data by interpolation is computationally slow. May I know how you changed the frequency of this part of the data from 3 hours to 1 hour.

**Reply by patrick-zippenfenig** [reply]
yeah, spline does not work. you have to use a solar model to calculate extraterrestrial radiation. It also needs to be backwards averaged. Typically you habe to super sample it at 1 to 5 min resolution and then average it. Afterwards you can calculate the clearness index ‘kt’, use spline or hermite interpolation and then calculate the shortwave radiation afterwards.

Maybe in python the pvlib can help. I usually just implement all the formulas myself. You can search for zensun in this repo.

Unfortunately, all calculations are cpu intensive and it is hard to implement them efficiently
